File: tests/Pieces/IndirectTest/java/AccessInvokeTresholdExceeded/src/MyTest.java
import org.junit.TestCase;

// use treshold = 4

public class MyTest extends TestCase {

    public void testCommand() {
        A a = new A();
        B b = new B();
        C c = new C();
        D d = new D();
        E e = new E();
        F f = new F();

        a.a();
        b.b();
        c.c();

        int i;
        i = d.d;
        i = e.e;
        i = f.f;
    }

}

File: tests/Pieces/IndirectTest/cpp/AccessInvokeTresholdExceeded/src/MyTest.cpp
#include "cppunit/TestSuite.h"
#include "cppunit/TestCaller.h"
#include "MyTest.h"
#include "Stuff.h"

using CppUnit::TestSuite;
using CppUnit::TestCaller;

void MyTest::testCommand() {
    A a;
    B b;
    C c;
    a.a();
    b.b();
    c.c();

    D d;
    E e;
    F f;
    int i;
    i = d.d;
    i = e.e;
    i = f.f;
}

Test* MyTest::suite() {
    TestSuite *suite = new TestSuite();
    suite->addTest(new TestCaller<MyTest>("testCommand", &MyTest::testCommand));
    return suite;
}
